packages/cli/src/index.ts

```diff
@@ -45,133 +82,51 @@ const handleError = (err: Error) => {
-  const argsList = (this._args as Array<{required: boolean; name: string}>)
-    .map((arg) => (arg.required ? `<${arg.name}>` : `[${arg.name}]`))
-    .join(' ');
-
-  if (this._alias) {
-    cmdName = `${cmdName}|${this._alias}`;
-  }
-
-  const sourceInformation = pkg
-    ? [`${chalk.bold('Source:')} ${pkg.name}@${pkg.version}`, '']
-    : [];
-
-  let output = [
-    chalk.bold(`react-native ${cmdName} ${argsList}`),
-    this._description ? `\n${this._description}\n` : '',
-    ...sourceInformation,
-    `${chalk.bold('Options:')}`,
-    this.optionHelp().replace(/^/gm, '  '),
-  ];
-
-  if (examples && examples.length > 0) {
-    const formattedUsage = examples
-      .map((example) => `  ${example.desc}: \n  ${chalk.cyan(example.cmd)}`)
-      .join('\n\n');
 
-    output = output.concat([chalk.bold('\nExample usage:'), formattedUsage]);
-  }
-
-  return output.join('\n').concat('\n');
-}
-
-function printUnknownCommand(cmdName: string) {
-  const availableCommands = commander.commands.map((cmd: any) => cmd._name);
-  const suggestion = availableCommands.find((cmd: string) => {
-    return leven(cmd, cmdName) < cmd.length * 0.4;
-  });
-  let errorMsg = `Unrecognized command "${chalk.bold(cmdName)}".`;
-  if (suggestion) {
-    errorMsg += ` Did you mean "${suggestion}"?`;
-  }
-  if (cmdName) {
-    logger.error(errorMsg);
-    logger.info(
-      `Run ${chalk.bold(
-        '"react-native --help"',
-      )} to see a list of all available commands.`,
-    );
-  } else {
-    commander.outputHelp();
-  }
+  process.exit(1);
 }
 
 /**
- * Custom type assertion needed for the `makeCommand` conditional
- * types to be properly resolved.
- */
-const isDetachedCommand = (
-  command: Command<boolean>,
-): command is Command<true> => {
-  return command.detached === true;
-};
-
-/**
- * Attaches a new command onto global `commander` instance.
- *
- * Note that this function takes additional argument of `Config` type in case
- * passed `command` needs it for its execution.
+ * Registers a new command within Commander interface
  */
-function attachCommand<IsDetached extends boolean>(
-  command: Command<IsDetached>,
-  ...rest: IsDetached extends false ? [Config] : []
-): void {
-  const options = command.options || [];
-  const cmd = commander
-    .command(command.name)
-    .action(async function handleAction(
-      this: commander.Command,
-      ...args: string[]
-    ) {
-      const passedOptions = this.opts();
-      const argv = Array.from(args).slice(0, -1);
-
-      try {
-        assertRequiredOptions(options, passedOptions);
-        if (isDetachedCommand(command)) {
-          await command.func(argv, passedOptions);
-        } else {
-          await command.func(argv, rest[0] as Config, passedOptions);
-        }
-      } catch (error) {
-        handleError(error);
-      }
-    });
+function registerCommand(...[command, config]: [command: DetachedCommand] | [command: Command, config: Config]): commander.Command {
```

Comment:
Our ESLint parser is too low to handle this syntax
```suggestion
function registerCommand(...[command, config]: [DetachedCommand] | [Command, Config]): commander.Command {
```
Would be cool to upgrade in a followup